pypowerwall/local/pypowerwall_local.py

```diff
@@ -207,6 +207,65 @@ def poll(self, api: str, force: bool = False,
             # should be already a dict in cache, so just return it
             return payload
 
+    def post(self, api: str, payload: Optional[dict], din: Optional[str],
+             recursive: bool = False, raw: bool = False) -> Optional[Union[dict, list, str, bytes]]:
+
+        # We probably should not cache responses here
+        # Also, we may have to use different HTTP Methods such as POST, PUT, PATCH based on Powerwall API requirements
+        # For now we assume it's taking POST calls
+
+        url = "https://%s%s" % (self.host, api)
+        try:
+            if self.authmode == "token":
+                r = self.session.post(url, headers=self.auth, json=payload, verify=False, timeout=self.timeout,
+                                      stream=raw)
+            else:
+                r = self.session.post(url, cookies=self.auth, json=payload, verify=False, timeout=self.timeout,
+                                      stream=raw)
+        except requests.exceptions.Timeout:
+            log.debug('ERROR Timeout waiting for Powerwall API %s' % url)
+            return None
+        except requests.exceptions.ConnectionError:
+            log.debug('ERROR Unable to connect to Powerwall at %s' % url)
+            return None
+        except Exception as exc:
+            log.debug('ERROR Unknown error connecting to Powerwall at %s: %s' % (url, exc))
+            return None
+        if r.status_code == 404:
+            log.debug('404 Powerwall API not found at %s' % url)
+            return None
+        if 400 <= r.status_code < 500:
```

Comment:
In local mode any attempt to send payload to `/api/operation` results in 403. Should we treat it here as Unauthorized instead of re-attempting to re-login?

Reply:
Yes! Good idea.